components/micro-gateway-core/src/main/ballerina/gateway/services/token_revocation_event_listener.bal

```diff
@@ -0,0 +1,95 @@
+        printDebug(KEY_TOKEN_REVOCATION_JMS,"token revoked jms Message Received");
+        map<any>|error mapMessage=message.getMapMessageContent();
+        map<string> inputMap={};
+        if (mapMessage is map<any>) {
+            if (mapMessage.hasKey("revokedToken") && mapMessage.hasKey("ttl")) {
+                string revokedToken = <string>mapMessage["revokedToken"];
+                string ttl= <string>mapMessage["ttl"];
+                inputMap[revokedToken]=ttl;
+                var status= addToRevokedTokenMap(inputMap);
+                if(status is boolean){
+                    printDebug(KEY_TOKEN_REVOCATION_JMS, "Successfully added to revoked token map");
+                }else{
+                    printDebug(KEY_TOKEN_REVOCATION_JMS, "Error while ading revoked token to map");
+                }
+            }else{
+                printDebug(KEY_TOKEN_REVOCATION_JMS, "No keys named revokedToken and ttl");
+            }
+
+        } else {
+            printError(KEY_TOKEN_REVOCATION_JMS, "Error occurred while reading message");
+        }
+    }
+};
+
+# `startSubscriberService` function create jms connection, jms session and jms topic subscriber.
+# It binds the subscriber endpoint and jms listener
+#
+# + return - jms:TopicSubscriber for token Revocation
+public function startTokenRevocationSubscriberService() returns jms:TopicSubscriber {
+    // Initialize a JMS connectiontion with the provider.
+    jms:Connection jmsTokenRevocationConnection = new({
+            initialContextFactory: jmsConnectioninitialContextFactoryTokenRevocation,
+            providerUrl: jmsConnectionProviderUrlTokenRevocation,
+            username: jmsConnectionUsernameTokenRevocation,
+            password: jmsConnectionPasswordTokenRevocation
+        });
+    // Initialize a JMS session on top of the created connection.
+    jms:Session jmsTokenRevocationSession = new(jmsTokenRevocationConnection, {
+            acknowledgementMode: "AUTO_ACKNOWLEDGE"
+        });
+
+    jms:TopicSubscriber subscriberTokenRevocationEndpoint = new(jmsTokenRevocationSession, topicPattern = "jwtRevocation");
+    _ = subscriberTokenRevocationEndpoint.__attach(jmsTokenRevocationListener, {});
+    _ = subscriberTokenRevocationEndpoint.__start();
+    return subscriberTokenRevocationEndpoint;
+}
+
+# `initiateTokenRevocationJmsListener` function initialize jmslistener subscriber service if `enabledTokenRevocation`
+# is enabled
+#
+# + return - boolean value of jmslistener started or not
+public function initiateTokenRevocationJmsListener() returns boolean {
+    boolean enabledRealtimeMessage = getConfigBooleanValue(REALTIME_MESSAGE_INSTANCE_ID,
+        REALTIME_MESSAGE_ENABLED, false);
+
+    if (enabledRealtimeMessage) {
+        jms:TopicSubscriber topicTokenRevocationSubscriber = startTokenRevocationSubscriberService();
+        printInfo(KEY_TOKEN_REVOCATION_JMS , "subscriber service for token revocation is started");
```

Comment:
Formatting issue. Shall we only keep a space after the comma? not before?

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/887